plugins/mac-avcapture/OBSAVCapture.m

```diff
@@ -475,34 +419,64 @@ - (BOOL)configureSession:(NSError *__autoreleasing *)error
 
     CMTime time = {.value = fps.denominator, .timescale = fps.numerator, .flags = 1};
 
-    AVCaptureDeviceFormat *format = nil;
-
-    for (AVCaptureDeviceFormat *formatCandidate in [self.deviceInput.device.formats reverseObjectEnumerator]) {
-        CMVideoDimensions formatDimensions = CMVideoFormatDescriptionGetDimensions(formatCandidate.formatDescription);
+    const char *selectedFormat = obs_data_get_string(self.captureInfo->settings, "supported_format");
+    NSString *selectedFormatNSString = [NSString stringWithCString:selectedFormat encoding:NSUTF8StringEncoding];
 
-        if (!(formatDimensions.width == dimensions.width) || !(formatDimensions.height == dimensions.height)) {
-            continue;
-        }
+    AVCaptureDeviceFormat *format = nil;
+    FourCharCode subtype;
+    OBSAVCaptureColorSpace colorSpace;
+    bool fpsSupported = false;
 
-        for (AVFrameRateRange *range in formatCandidate.videoSupportedFrameRateRanges) {
-            if (CMTimeCompare(range.maxFrameDuration, time) >= 0 && CMTimeCompare(range.minFrameDuration, time) <= 0) {
+    if (![selectedFormatNSString isEqual:@""]) {
```

Comment:
If you know that both objects are `NSString`, it is at least measurably faster to call `isEqualToString`.

Reply:
Changed.